packages/fxa-auth-db-mysql/lib/db/mysql.js

```diff
@@ -1192,34 +1192,29 @@ module.exports = function(log, error) {
         // Enforce sane defaults on every new connection.
         // These *should* be set by the database by default, but it's nice
         // to have an additional layer of protection here.
-        connection.query('SELECT @@sql_mode AS mode;', (err, rows) => {
-          if (err) {
-            return reject(err);
-          }
-
-          const currentModes = rows[0]['mode'].split(',');
-          this.requiredModes.forEach(requiredMode => {
-            if (currentModes.indexOf(requiredMode) === -1) {
-              currentModes.push(requiredMode);
-            }
-          });
-
-          const newMode = currentModes.join(',');
-          connection.query(`SET SESSION sql_mode = '${newMode}';`, err => {
-            if (err) {
-              return reject(err);
-            }
-
-            connection.query('SET NAMES utf8mb4 COLLATE utf8mb4_bin;', err => {
-              if (err) {
-                return reject(err);
+        const query = P.promisify(connection.query, { context: connection });
+        return resolve(
+          (async () => {
```

Comment:
There was no way I was going to add another layer to this pyramid of doom; happy to take suggestions on a nicer way to async-await-ify this code without this turning into a whole-class refactor.